cmake/bintools/llvm/target_bintools.cmake

```diff
@@ -12,9 +12,6 @@
 #   elfconvert_flag_section_remove: --remove-section=
 #   elfconvert_flag_section_only  : --only-section=
 #   elfconvert_flag_section_rename: --rename-section;
-#   elfconvert_flag_gapfill       : --gap-fill;
```

Comment:
The `--gap-fill` option was added into `llvm-objdump` fairly recently (in Dec. 2023, please see https://github.com/llvm/llvm-project/commit/4070dffd34e99915b005c655086d92e42c004d25) and is available in the official LLVM 18 release at this point.

While I understand that most people are probably not using the latest clang releases, I'd prefer to see this support remain in Zephyr as `--gap-fill` support is present in an official LLVM release and will be present in future releases. I do know of a few users who are on LLVM-18-based toolchains and are using it to build Zephyr (and have `--gap-fill` enabled!) so I'm hoping we can avoid breaking those who are on the latest tools.